### PR TITLE
[CM-1344] Added support for custom input field rich message

### PR DESCRIPTION
**Pull request by himanshujoshi848**
## Summary
Added support for custom input field rich message in IOS SDK.

## How it works ?

- For payload and bot setup you can use the following [data](https://docs.kommunicate.io/docs/message-types#custom-input-field)

-  Response from bot

Simulator Screenshot - iPhone 14 Pro - 2023-09-11 at 00 19 45

- Updated dashboard
Screenshot 2023-09-11 at 12 20 20 AM

On updating the OTP, it updated the OTP as user's metadata and it got reflected in user info column

### CODE REVIEW

File: Sources/Controllers/ALKConversationViewController.swift [hunk 1]
```diff
@@ -2050,6 +2058,29 @@ open class ALKConversationViewController: ALKBaseViewController, Localizable {
...
+    
+    public func showInvalidReplyAlert(kmField: KMField) {
+        
+        guard let validation = kmField.validation else {
+            return
+        }
+        
+        let title = "Invalid Reply"
+        let message = validation["errorText"]
+        let okButtonTitle = "OK"
```
Comment (Sathyan-KM) [inline Sources/Controllers/ALKConversationViewController.swift:2070]: add localization for this texts

File: Sources/Models/ALKMessageModel.swift [hunk 2]
```diff
@@ -180,3 +180,62 @@ public extension ALKMessageViewModel {
...
+                return nil
+            }
+            let kmFieldData = payload.data(using: .utf8)
+            guard let kmFieldData = kmFieldData else {
+                return nil
+            }
+            let kmField = try JSONDecoder().decode(KMField.self, from: kmFieldData)
+            return kmField
+        } catch {
+            print("Error decoding JSON: \(error)")
```
Comment (Sathyan-KM) [inline Sources/Models/ALKMessageModel.swift:212]: need clear log here

File: Sources/ViewModels/ALKConversationViewModel.swift [hunk 3]
```diff
@@ -771,6 +782,86 @@ open class ALKConversationViewModel: NSObject, Localizable {
...
     }
     
+    func updateMetaDataForCustomField(message : ALMessage){
+        if let lastMessage = alMessages.last?.messageModel,
+           let replyMetaData = lastMessage.getReplyMetaData() {
+            message.metadata.addEntries(from: replyMetaData)
+        }
+    }
+    
+    func updateUser(kmField: KMField, message: String) {
```
Comment (Sathyan-KM) [inline Sources/ViewModels/ALKConversationViewModel.swift:792]: Why are we naming it as message:String . Are we passing message here

Comment (himanshujoshi848) [reply Sources/ViewModels/ALKConversationViewModel.swift:792]: yes the message that the user sent we are passing that to update the userDetails

**Review by Sathyan-KM — APPROVED** [review]

**Pull request merged**